src/couchapps/ReqMgr/updates/updaterequest.js

```diff
@@ -79,11 +92,14 @@ function(doc, req) {
     	} else {
     		doc[key] = newValues[key];
     	}
-       
+       	
         // If key is RequestStatus, also update the transition
         if (key == "RequestStatus") {
         	updateTransition();
         }
+        
+        updateTaskStepChain("Task", key, newValues[key]);
+        updateTaskStepChain("Step", key, newValues[key]);
```

Comment:
Instead of calling this function twice, how about we iterate over ["Task", "Step"] inside the function itself? It should have a slightly better performance.